python/triton/runtime/jit.py

```diff
@@ -485,6 +485,8 @@ class JITFunction(KernelInterface[T]):
     # Hook to signal that a kernel is done compiling and inspect compiled function.
     # cache_hook will always be called before compilation and compiled_hook after.
     compiled_hook = None
+    # Default run hooks - provide means of overriding run hooks globally
+    default_prerun_hooks = []
```

Comment:
One passing thought: it seems everything that inherits from `KernelInterface` has some concept of pre/post hook-- what do you think about unifying these hooks onto that class directly? Of course I think some care will be needed to not break current usages, though.